src/App.js

```diff
@@ -1,33 +1,122 @@
+import { useState } from 'react'
 import "./App.css";
 import "./form.css";
 
+function Header({ children }) {
+  return (<div>{children}</div>)
+}
+
+function Subheader({ children }) {
+  return (<div>{children}</div>)
+}
+
+function Footer({ children }) {
+  return (<div>{children}</div>)
+}
+
+function Button({ type, label, disabled }) {
+  return (<button disabled={disabled} type={type}>{label}</button>)
+}
+
+
+
+function Field({ type, label, htmlFor, value, onChange }) {
+  return (
+    <div>
+      <label htmlFor={htmlFor}>{label}:</label>
+      <input id={htmlFor} type={type} value={value} onChange={onChange} />
+    </div>
+  )
+}
+
+const listaCopy = ['One lowercase character', 'One uppercase character', 'One number', 'One special character', '8 characters minimum']
+const PasswordRequirements = () => {
+  return (
+    <ul class='reqList'>
+      {listaCopy.map(copy => (
+        <li>{copy}</li>
+      ))}
+    </ul>
+  )
+}
+
 function App() {
+  const [email, setEmail] = useState('')
+  const [userName, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [emailError, emailSetError] = useState(null)
+  const [usernameError, usernameSetError] = useState(null)
+  const [passwordError, passwordSetError] = useState(null)
+
+  const handleEmailChange = (e) => {
+    const email = e.target.value
+    const containsAt = email.includes('@')
+    emailSetError(containsAt ? null : 'Email is not vaild')
+    console.log(containsAt)
+    setEmail(email)
+  }
+
+  const handleUsernameChange = (e) => {
+    const userName = e.target.value
+    const isLowerCase = userName === userName.toLowerCase()
+    usernameSetError(isLowerCase ? null : 'Username must be lower case')
+    console.log(isLowerCase)
+    setUsername(userName)
+  }
+
+  const handlePasswordChange = (e) => {
+    const password = e.target.value
+    var regex = /^(?=.*\d)(?=.*[!@#$%^&*])(?=.*[a-z])(?=.*[A-Z]).{8,}$/;
```

Comment:
I can see this is copy/paste because of usage of `var`.
Regexes like this are very hard to read. I would suggest to split it and handle it case by case see https://github.com/livechat/fun-with-flags/pull/1/files#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67R64